### ISSUE TITLE
Value Checker gives up on unsigned right shifts for ranges

### ISSUE THREAD

**Issue by panacekcz** [typetools/checker-framework#1655]
In cases similar to #1623, the Value Checker does not infer the value range when an unsigned right shift is used.
Code:
```java
import org.checkerframework.common.value.qual.IntRange;

public class Shifts {
	public void m(int a) {
		@IntRange(from = 0, to = 255) int b = a & 0xff;
		
		@IntRange(from = 0, to = 15) int c1 = b >> 4; // OK
		@IntRange(from = 0, to = 15) int c2 = b >>> 4; // False positive
	}
}
```
Output of the Value Checker:
```
Shifts.java:8: error: [assignment.type.incompatible] incompatible types in assignment.
		@IntRange(from = 0, to = 15) int c2 = b >>> 4; // False positive
		                                        ^
  found   : @IntRange(from=-2147483648, to=2147483647) int
  required: @IntRange(from=0, to=15) int
1 error
```
Expected: no warnings.

Encountered in Guava. Using unsigned shift in such case might be a "code style" choice, because the result is the same as with signed shifts.

**Comment by mernst** [comment]
@JasonMrX , would it be possible for you to take a look at this?  Thanks!

**Comment by JasonMrX** [comment]
@mernst I will take a look. Thanks.

**Comment by JasonMrX** [comment]
Consider the following case:
```java
import org.checkerframework.common.value.qual.IntRange;
public class Shifts {
    public void m(int a) {
        @IntVal({2147483647}) long a1 = -1 >>> 1;
        @LongVal({9223372036854775807}) long a2 = -1L >>> 1;
    }
}
```
Both `-1` and `-1L` are having the same `Range` representation -- `new Range(-1, -1)`, however, the refined range after the unsigned shift is different. This is due to the nature of shift operation that it's actually type dependent. Therefore if we want this behavior in the `Range.java` library, we need to bring this type information into it, which is something that we were trying to avoid in the first place (to keep the `Range` library away from the underlying type but only focus on the range calculation itself)

If we really want this behavior, I would suggest adding methods like `unsignedRightShiftForInt` and `unsignedRightShiftForLong` to the `Range` library and call them accordingly in the value checker based on the type of the left operand. Any suggestions? @panacekcz  @mernst 

Thanks,

Jason

**Comment by panacekcz** [comment]
That sounds like a reasonable solution. However, for me now it would be sufficient if it worked for non-negative numbers.
